src/components/vm/nics/vmNicsCard.jsx

```diff
@@ -359,7 +361,7 @@ export class VmNetworkTab extends React.Component {
                         objectType: _("network interface"),
                         objectName: network.mac,
                         onClose: () => this.setState({ deleteDialogProps: undefined }),
-                        deleteHandler: () => domainDetachIface({ connectionName: vm.connectionName, mac: network.mac, vmName: vm.name, live: vm.state === 'running', persistent: vm.persistent }),
+                        deleteHandler: () => domainDetachIface({ connectionName: vm.connectionName, mac: network.mac, vmName: vm.name, live: vm.state === 'running', persistent: nicPersistent }),
```

Comment:
I don't understand this change here, we used to check if the VM is not persistent, so that we do --no-define when running the virt-xml, That sounds correct, why you change this?

Reply:
That would not work in a situation when a VM is persistend, but the vNIC is not. When you have a running VM, and during attachment of vNIC you decide to not pick `Always attach` option, then vNIC is defined only in live XML. 
Therefore when removing such vNIC, not calling --no-define would result in an error.
Therefore for detaching vNICs, it's the persistence of vNIC we need to look at